pkg/kube/inject/webhook.go

```diff
@@ -731,12 +731,21 @@ func getDeployMetaFromPod(pod *corev1.Pod) (*metav1.ObjectMeta, *metav1.TypeMeta
 			typeMetadata.Kind = controllerRef.Kind
 
 			// heuristic for deployment detection
+			deployMeta.Name = controllerRef.Name
 			if typeMetadata.Kind == "ReplicaSet" && pod.Labels["pod-template-hash"] != "" && strings.HasSuffix(controllerRef.Name, pod.Labels["pod-template-hash"]) {
 				name := strings.TrimSuffix(controllerRef.Name, "-"+pod.Labels["pod-template-hash"])
 				deployMeta.Name = name
 				typeMetadata.Kind = "Deployment"
-			} else {
-				deployMeta.Name = controllerRef.Name
+			} else if typeMetadata.Kind == "Job" && len(controllerRef.Name) > 11 {
+				// If job name suffixed with `-<ten-digit-timestamp>`, trim the suffix and set kind to cron job.
```

Comment:
Is it always 10?

Reply:
It should be always 10 digit epoch: https://github.com/kubernetes/kubernetes/blob/0fd10997dfee0f9ec2c413004bf93d310d76c84b/pkg/controller/cronjob/utils.go#L154

I cannot find this is documented anywhere, but I think it is unlikely to change.